mkt/site/tests/__init__.py

```diff
@@ -344,7 +346,99 @@ def login(self, profile):
+
+    Fixtures are loaded once per class, and a class setUpTestData method is
+    added to be overridden by sublasses.  `transaction.atomic()` is used to
+    achieve test isolation.
+    See orginal code:
+    https://github.com/django/django/blob/1.8b2/django/test/testcases.py
+    #L747-990.
+
+    A noteable difference is that this class assumes the database supports
+    transactions.  This class will be obsolete on upgrade to 1.8.
+    """
+    fixtures = None
+
+    @classmethod
+    def _databases_names(cls, include_mirrors=True):
+        # If the test case has a multi_db=True flag, act on all databases,
+        # including mirrors or not. Otherwise, just on the default DB.
+        if getattr(cls, 'multi_db', False):
+            return [alias for alias in connections
+                    if (include_mirrors or
+                        connections[alias].settings_dict['TEST']['MIRROR'])]
+        else:
+            return [DEFAULT_DB_ALIAS]
+
+    @classmethod
+    def _enter_atomics(cls):
+        """Helper method to open atomic blocks for multiple databases"""
+        atomics = {}
+        for db_name in cls._databases_names():
+            atomics[db_name] = transaction.atomic(using=db_name)
+            atomics[db_name].__enter__()
+        return atomics
+
+    @classmethod
+    def _rollback_atomics(cls, atomics):
+        """Rollback atomic blocks opened through the previous method"""
+        for db_name in reversed(cls._databases_names()):
+            transaction.set_rollback(True, using=db_name)
+            atomics[db_name].__exit__(None, None, None)
+
+    @classmethod
+    def setUpClass(cls):
+        cls.cls_atomics = cls._enter_atomics()
+
+        if cls.fixtures:
+            for db_name in cls._databases_names(include_mirrors=False):
+                    try:
+                        call_command('loaddata', *cls.fixtures, **{
+                            'verbosity': 0,
+                            'commit': False,
+                            'database': db_name,
+                        })
+                    except Exception:
+                        cls._rollback_atomics(cls.cls_atomics)
+                        raise
+        cls.setUpTestData()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._rollback_atomics(cls.cls_atomics)
+        for conn in connections.all():
+            conn.close()
+
+    @classmethod
+    def setUpTestData(cls):
+        """Load initial data for the TestCase"""
+        pass
+
+    def _should_reload_connections(self):
+        return False
+
+    def _fixture_setup(self):
+        assert not self.reset_sequences, (
+            'reset_sequences cannot be used on TestCase instances')
+        self.atomics = self._enter_atomics()
+
+    def _fixture_teardown(self):
+        self._rollback_atomics(self.atomics)
+
+    def _post_teardown(self):
+        # Monkey-patch so connections don't get closed
```

Comment:
That's an original hack/monkeypatch :)
In 1.6 django's _post_teardown connections are closed and we don't want that to happen after each test anymore. https://github.com/django/django/blob/1.6.10/django/test/testcases.py#L788